sklearn/neural_network/tests/test_mlp.py

```diff
@@ -432,7 +432,8 @@ def test_predict_proba_binary():
     X = X_digits_binary[:50]
     y = y_digits_binary[:50]
 
-    clf = MLPClassifier(hidden_layer_sizes=5)
+    clf = MLPClassifier(hidden_layer_sizes=5,
+                        random_state=np.random.RandomState())
```

Comment:
This should be set to an int or to a fixed `RandomState` instance e.g. `np.random.RandomState(0)`.

But whether a test passes or fails shouldn't depend on a random seed (reminds me of https://github.com/scikit-learn/scikit-learn/pull/12624#issuecomment-440603562).

This tests checks a ROC AUC score while ignoring `ConvergenceWarning` in `fit()`. We should instead make sure the classifier converges by using the right parameters.

Reply:
In this case changing the activation to 'logistic' passes 100 random seeds.

```
@pytest.mark.parametrize('random_state', range(100))
def test_predict_proba_binary(random_state):
    # Test that predict_proba works as expected for binary class.
    X = X_digits_binary[:50]
    y = y_digits_binary[:50]

    clf = MLPClassifier(hidden_layer_sizes=5, random_state=random_state,
                        activation='logistic')
    clf.fit(X, y)
...
```

So I would say the correct fix is to use

```
    clf = MLPClassifier(hidden_layer_sizes=5, activation='logistic')
```

and not ignore the `ConvergeneWarning` while fitting.

Reply:
I see. I will update it. Thanks.